lib/runner.js

```diff
@@ -597,12 +599,23 @@ Runner.prototype.runSuite = function(suite, fn) {
 
   function done(errSuite) {
     self.suite = suite;
-    self.hook('afterAll', function() {
-      self.emit('suite end', suite);
+    self.nextSuite = next;
+
+    if (afterAllHookCalledOnce) {
```

Comment:
Some small nitpicks that I'll fix in the merge if you don't have time:
Renaming afterAllHookCalledOnce => afterAllCalled or afterAllHookCalled